include/seqan3/io/sequence/sequence_file_format_fastq.hpp

```diff
@@ -0,0 +1,326 @@
+#include <seqan3/io/detail/ignore_output_iterator.hpp>
+#include <seqan3/io/detail/output_iterator_conversion_adaptor.hpp>
+#include <seqan3/io/sequence/sequence_file_in_options.hpp>
+#include <seqan3/io/sequence/sequence_file_out_options.hpp>
+#include <seqan3/io/stream/parse_condition.hpp>
+#include <seqan3/range/detail/misc.hpp>
+#include <seqan3/range/view/char_to.hpp>
+#include <seqan3/range/view/to_char.hpp>
+#include <seqan3/range/view/take_line.hpp>
+#include <seqan3/std/concept/range.hpp>
+#include <seqan3/std/view/subrange.hpp>
+#include <seqan3/std/view/transform.hpp>
+
+namespace seqan3
+{
+/*!\brief       The FastQ format.
+ * \implements  sequence_file_format_concept
+ * \ingroup     sequence
+ *
+ * \details
+ *
+ * ### Introduction
+ *
+ * FastQ is the de-facto-standard for storing sequences together with quality information. See the
+ * [article on wikipedia](https://en.wikipedia.org/wiki/FASTQ_format) for a an in-depth description of the format.
+ *
+ * ### Fields
+ *
+ * The FastQ format provides the fields seqan3::field::SEQ, seqan3::field::ID and seqan3::field::QUAL; or alternatively
+ * provides seqan3::field::SEQ_QUAL as a single field of sequence and quality. All three fields (or ID + SEQ_QUAL) are
+ * required when writing and the sequence and qualities are required to be of the same length.
+ *
+ * ### Encodings
+ *
+ * All documented encodings for the quality string are supported (see the article above), but they are **not detected**
+ * from the file. Instead, when reading the file, you have to set the respective alphabet via a traits type (see
+ * seqan3::sequence_file_in_traits_concept and the quality submodule \todo link).
+ *
+ * ### Implementation notes
+ *
+ * This implementation supports the following optional features of the format:
+ *
+ *   * line breaks and/or other whitespace characters in any part of the sequence and/or qualities (only when reading!)
+ *   * writing the ID to the `+`-line also (line is always ignored when reading)
+ *
+ */
+class sequence_file_format_fastq
+{
+public:
+    /*!\name Constructors, destructor and assignment
+     * \brief Rule of five explicitly defaulted.
+     * \{
+     */
+    sequence_file_format_fastq() = default;
+    sequence_file_format_fastq(sequence_file_format_fastq const &) = delete;
+    sequence_file_format_fastq & operator=(sequence_file_format_fastq const &) = delete;
+    sequence_file_format_fastq(sequence_file_format_fastq &&) = default;
+    sequence_file_format_fastq & operator=(sequence_file_format_fastq &&) = default;
+    //!\}
+
+    //!\brief The valid file extensions for this format; note that you can modify this value.
+    static inline std::vector<std::string> file_extensions
+    {
+        { "fastq" },
+        { "fq"    }
+    };
+
+    //!\copydoc sequence_file_in_format_concept::read
+    template <typename stream_type,     // constraints checked by file
+              typename seq_legal_alph_type, bool seq_qual,
```

Comment:
in the concept you call it `seq_qual_in_one`, please change either of these for consistency